CMakeLists.txt

```diff
@@ -460,6 +464,62 @@ else()
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-strict-aliasing")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-strict-aliasing")
 
+  # if those don't work for your compiler, single it out where appropriate
+  if(CMAKE_BUILD_TYPE EQUAL "Release")
+    set(C_SECURITY_FLAGS "${C_SECURITY_FLAGS} -D_FORTIFY_SOURCE=2")
+    set(CXX_SECURITY_FLAGS "${CXX_SECURITY_FLAGS} -D_FORTIFY_SOURCE=2")
+  endif()
+
+  # warnings
+  check_cxx_compiler_flag(-Wformat HAS_CXX_WFORMAT)
```

Comment:
These are kinda wordy... how about:

```
function(add_if_cxx_flag_exists FLAG FLAG_LIST)
  check_cxx_compiler_flag(${FLAG} HAS_FLAG)
  if(HAS_FLAG)
    set(FLAG_LIST "${FLAG_LIST} ${FLAG}")
  endif()
endfunction()

# likewise for add_if_c_flag_exists

add_if_cxx_flag_exists(-Wformat CXX_SECURITY_FLAGS)
add_if_cxx_flag_exists(-Wformat-security CXX_SECURITY_FLAGS)
add_if_cxx_flag_exists(-fstack-protector CXX_SECURITY_FLAGS)
add_if_cxx_flag_exists(-fstack-protector-strong CXX_SECURITY_FLAGS)

# likewise, again
```

No need to do the if/else logic on the strong stack protector flag either, since the compiler will just pick the last one

Reply:
cmake is such a son of a bitch...
Those three funcs might be mergeable, if the stock macro can somehow be passed as a parm too